### PR TITLE
Add message publishing to payer service

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Actually publishes messages to nodes and blockchain from payer service
- Changes to write ClientEnvelopes to chain instead of underlying payload types

## Next up

- Fix blockchain reading to match published message types
- Add tests

## AI Assisted summary

This PR implements the logic for publishing messages to nodes and the blockchain from the payer service. It moves the responsibility of publishing to the blockchain from the message service to the payer service. The payer service now groups incoming envelopes by their destination (node or blockchain) and handles the publishing accordingly.

The blockchain publishing interface has been updated to return transaction hashes, allowing for better tracking and verification of published messages. The code now writes entire ClientEnvelopes to the blockchain instead of just the underlying payload, providing more context and consistency in the data storage approach.

Additionally, the PR includes updates to related test files and mock interfaces to accommodate these changes, ensuring the new functionality is properly tested and integrated.

### CODE REVIEW

**Comment by neekolas** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/245?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#246** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/246?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#245** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/245?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#243** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/243?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#241** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/241?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#239** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/239?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#238** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/238?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#237** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/237?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by mkysel — APPROVED** [review]

**Comment by neekolas** [comment]
### Merge activity

* **Oct 21, 3:52 PM PDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/xmtp/xmtpd/245).
